apps/api/src/app/events/usecases/add-job/add-delay-job.usecase.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@nestjs/common';
+import { JobEntity, JobRepository, JobStatusEnum } from '@novu/dal';
+import { DigestUnitEnum, StepTypeEnum, DelayTypeEnum } from '@novu/shared';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { WorkflowQueueService } from '../../services/workflow.queue.service';
+import { AddJobCommand } from './add-job.command';
+import { differenceInMilliseconds } from 'date-fns';
+
+@Injectable()
+export class AddDelayJob {
+  constructor(private jobRepository: JobRepository, private workflowQueueService: WorkflowQueueService) {}
+
+  public async execute(command: AddJobCommand): Promise<boolean> {
+    const data = await this.jobRepository.findById(command.jobId);
+
+    if (!data) {
+      return false;
+    }
+
+    const isDelayStep = data.type === StepTypeEnum.DELAY;
+
+    if (!isDelayStep) {
+      return false;
+    }
```

Comment:
```suggestion
    const isDelayStep = data?.type === StepTypeEnum.DELAY;

    if (!data || !isDelayStep) {
      return false;
    }
```
A grouping we could do in this internal validation we do.
We could even remove the need of checking if data exists as it would be included in `isDelayStep`.